src/setup.jl

```diff
@@ -153,8 +153,12 @@ function locate_libsingular()
    # by comparing tree hashes
    src_hash = bytes2hex(Pkg.GitTools.tree_hash(joinpath(@__DIR__, "..", "deps", "src")))
 
-   if jll_hash == src_hash
+   # the uuid is for Singular.jl
+   pkginfo = get(Pkg.dependencies(), Base.UUID("bcd08a7b-43d2-5ff7-b6d4-c458787f915c"), nothing)
```

Comment:
Instead of hardcoding the UUID could also query it
```suggestion
   pkginfo = get(Pkg.dependencies(), Base.PkgId(parentmodule(Setup)).uuid, nothing)
```